assets/src/components/map.tsx

```diff
@@ -461,27 +447,35 @@ export const BaseMap = (props: Props): ReactElement<HTMLDivElement> => {
             trainVehicle={trainVehicle}
           />
         ))}
-
         {(props.shapes || []).map((shape) => (
           <RouteShape key={shape.id} shape={shape} />
         ))}
-        <RouteStopMarkers
-          stops={(props.shapes || []).flatMap((shape) => shape.stops || [])}
-          iconSize={stationIconSize}
-          direction={props.stopCardDirection}
-          includeStopCard={
-            props.includeStopCard && inTestGroup(MAP_BETA_GROUP_NAME)
-          }
-        />
-        {zoomLevel >= 15 &&
-          props.stations?.map((station) => (
-            <StationMarker
-              key={station.id}
-              station={station}
-              iconSize={stationIconSize}
-            />
-          ))}
-        {zoomLevel >= 15 && <GarageMarkers zoomLevel={zoomLevel} />}
+
+        <ZoomLevelWrapper>
+          {(zoomLevel) => (
+            <>
+              <RouteStopMarkers
+                stops={(props.shapes || []).flatMap(
+                  (shape) => shape.stops || []
+                )}
+                zoomLevel={zoomLevel}
+                direction={props.stopCardDirection}
+                includeStopCard={
+                  props.includeStopCard && inTestGroup(MAP_BETA_GROUP_NAME)
+                }
+              />
+              {zoomLevel >= 15 &&
+                props.stations?.map((station) => (
+                  <StationMarker
+                    key={station.id}
+                    station={station}
+                    zoomLevel={zoomLevel}
+                  />
+                ))}
+              {zoomLevel >= 15 && <GarageMarkers zoomLevel={zoomLevel} />}
+            </>
+          )}
+        </ZoomLevelWrapper>
```

Comment:
I rebased myself off of this and I'm having trouble with `RouteStopMarkers` being provided always instead of if there's shapes available, as well as we're always going to have two `ZoomLevelWrappers` if this stays in `BaseMap`, should we maybe move this into a `DefaultMap`?.. I'm struggling to figure out the right layers to allow a good default map dev experience but also allow escaping out of default elements we provide that we don't always use. But we're getting closer and closer to `BaseMap` providing a few defaults that we like and otherwise being declaratively specified through `children`, which then could be wrapped to provide the existing functionality.

Reply:
I've been considering using leaflet-react's form of allowing disabling of default controls via props like `zoomControl={false}` but I've been trying to remove props from `BaseMap` lol

Reply:
like, we could add a `zoomElements: (zoomLevel) => ReactElement` and call that within the `ZoomLevelWrapper` provided in `BaseMap` but I think it'd be better to do that in a wrapper of `BaseMap` instead.

Reply:
Fully agree on your first point - I'll update to conditionally render <RouteStopMarkers/> only if there are actually stops to render. 

I don't totally mind the two separate ZoomLevelWrappers - it seems advantageous to me to have the garage rendering done by one of the lower level maps so that it is done consistently across all maps.